src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -0,0 +1,490 @@
+        _control = closeFirst ? _secondChild->_control : _firstChild->_control;
+        _profile = closeFirst ? _secondChild->_profile : _firstChild->_profile;
+
+        // If either of our children was focused, we want to take that focus from
+        // them.
+        _lastFocused = _firstChild->_lastFocused || _secondChild->_lastFocused;
+
+        // Remove all the ui elements of our children. This'll make sure we can
+        // re-attach the TermControl to our Grid.
+        _firstChild->_root.Children().Clear();
+        _secondChild->_root.Children().Clear();
+
+        // Reset our UI:
+        _root.Children().Clear();
+        _root.ColumnDefinitions().Clear();
+        _root.RowDefinitions().Clear();
+        _separatorRoot = { nullptr };
+
+        // Reattach the TermControl to our grid.
+        _AddControlToRoot(_control);
+
+        if (_lastFocused)
+        {
+            _control.GetControl().Focus(FocusState::Programmatic);
+        }
+
+        _splitState = SplitState::None;
+
+        // Release our children.
+        _firstChild = nullptr;
+        _secondChild = nullptr;
+    }
+    else
+    {
+        // Revoke the old event handlers.
+        _firstChild->Closed(_firstClosedToken);
+        _secondChild->Closed(_secondClosedToken);
+
+        // Steal all the state from our child
+        _splitState = remainingChild->_splitState;
+        _separatorRoot = remainingChild->_separatorRoot;
+        _firstChild = remainingChild->_firstChild;
+        _secondChild = remainingChild->_secondChild;
+
+        // Reset our UI:
+        _root.Children().Clear();
+        _root.ColumnDefinitions().Clear();
+        _root.RowDefinitions().Clear();
+
+        // Copy the old UI over to our grid.
+        // Start by copying the row/column definitions. Iterate over the
+        // rows/cols, and remove each one from the old grid, and attach it to
+        // our grid instead.
+        while (remainingChild->_root.ColumnDefinitions().Size() > 0)
+        {
+            auto col = remainingChild->_root.ColumnDefinitions().GetAt(0);
+            remainingChild->_root.ColumnDefinitions().RemoveAt(0);
+            _root.ColumnDefinitions().Append(col);
+        }
+        while (remainingChild->_root.RowDefinitions().Size() > 0)
+        {
+            auto row = remainingChild->_root.RowDefinitions().GetAt(0);
+            remainingChild->_root.RowDefinitions().RemoveAt(0);
+            _root.RowDefinitions().Append(row);
+        }
+
+        // Remove the child's UI elements from the child's grid, so we can
+        // attach them to us instead.
+        remainingChild->_root.Children().Clear();
+
+        _root.Children().Append(_firstChild->GetRootElement());
+        _root.Children().Append(_separatorRoot);
+        _root.Children().Append(_secondChild->GetRootElement());
+
+        // Set up new close handlers on the children, so thay'll notify us
+        // instead of their old parent.
+        _SetupChildCloseHandlers();

```

Comment:
Does some sort of lock need to be taken during this process to prevent a close from somehow happening while you're reshuffling the UI and the handlers are detached?

Reply:
That's a good point - lemme work on this

Reply:
I'm mildly worried about the fact that I can't really test this scenario. I have a fix in mind, but there's very little way for me to be able to test it currently.

Reply:
I added a follow-up to add tests #1124